baseapp/options.go

```diff
@@ -229,3 +229,14 @@ func (app *BaseApp) SetInterfaceRegistry(registry types.InterfaceRegistry) {
 	app.interfaceRegistry = registry
 	app.grpcQueryRouter.SetInterfaceRegistry(registry)
 }
+
+// SetStreamingService is used to set a streaming service into the BaseApp hooks and load the listeners into the multistore
+func (app *BaseApp) SetStreamingService(s StreamingService) {
+	// add the listeners for each StoreKey
+	for key, lis := range s.Listeners() {
+		app.cms.AddListeners(key, lis)
```

Comment:
```suggestion
		app.cms.AddListener(key, lis)
```

Reply:
This is a `map[types.Storekey][]store.WriteListeners` that is being iterated over, so it is potentially adding multiple listeners.